### PR TITLE
Add rule S4322 ('prefer-type-guard') for TS

### LINKED ISSUE #1519: Add rule S4322 ('prefer-type-guard' in tslint) for TS

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**
Fixes #1519

### CODE REVIEW

**Comment by quentin-jaquier-sonarsource** [comment]
In #1656, there is logic related to function that we could re-use, I propose to wait until it's merge on master to simplify the implementation of this one.

File: eslint-bridge/tests/rules/prefer-type-guard.test.ts [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        message: `Declare this function return type using type predicate "animal is Fish".`,
+        line,
+        endLine: line,
+      },
+    ],
+  };
+}
+
+ruleTester.run("Type guards should be used", rule, {
+  /*
```
Comment (yassin-kammoun-sonarsource) [inline eslint-bridge/tests/rules/prefer-type-guard.test.ts]: Do we still need those ?

File: eslint-bridge/src/rules/prefer-type-guard.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  const body = functionDeclaration.body;
+  const returnedExpression = getReturnedExpression(body);
+  if (!returnedExpression) {
+    return;
+  }
+
+  if (isInequalityBinaryExpression(returnedExpression)) {
+    const { left, right } = returnedExpression;
+    if (isUndefined(right)) {
```
Comment (christophe-zurn-sonarsource) [inline eslint-bridge/src/rules/prefer-type-guard.ts]: Why are we not checking the other way around too ? if `left` is `undefined`, why don't we check `right` expression whether it's a `checkCastType` ?

Comment (quentin-jaquier-sonarsource) [reply eslint-bridge/src/rules/prefer-type-guard.ts]: Good idea, it's a nice simple improvement to the rule!

**Pull request merged**